### ISSUE TITLE
Java Test Suite

### ISSUE THREAD

**Issue by wthit56** [Codewars/codewars.com#277]
A failing `assertEquals("A11y", abbr.abbreviate("Accessibility"));` seems be giving empty "expected" and "got" messages.

Kata in question: http://www.codewars.com/kata/5375f921003bf62192000746/discuss/javascript

**Comment by jhoffner** [comment]
Please file the issue within the discourse section of the kata. GitHub is for application related issues, not content related issues. 

Sent from Outlook Mobile




On Sun, Apr 10, 2016 at 10:19 AM -0700, "Thomas Giles" <notifications@github.com> wrote:












A failing assertEquals("A11y", abbr.abbreviate("Accessibility")); seems be giving empty "expected" and "got" messages.



Kata in question: http://www.codewars.com/kata/5375f921003bf62192000746/discuss/javascript

**Comment by wthit56** [comment]
I think you may have misunderstood the original post. That code is in a test within the kata.

The system's `assertEquals` method is not printing out the expected value when the test fails, even though it is provided as plain text.To me, this indicates a bug within the Java Test Suite, and _not_ with the kata itself. I expect the same problem will happen in any kata using the Java Test Suite.

The link was just for reference. I hope that makes sense?

**Comment by jhoffner** [comment]
ah ok. Thats odd, its such a simple low level method I'm not sure how it only be coming up now as an issue. I'm going to be focused on open sourcing all of this code this week so once I do, if I don't get to looking into this, someone in the community (or you) will be able to.

**Comment by paul-calvelage** [comment]
__Summary:__ I can reproduce this.  Kata discussion history leads me to believe this bug may have been present at least 1 year ago.  Inspection of the output shows that the expected and actual strings are not missing; rather, they are being mangled into HTML markup.

__History:__ A Codewars VIP, "mortonfox" (1 kyu, ranked 13 on CW), reported this as a content issue (apparently mistakenly) against the "Broken Greetings" Java translation approximately a year ago. See http://www.codewars.com/kata/broken-greetings/discuss/java.

__How to reproduce__: As it stands, it is enough to train on the "Broken Greetings" kata in Java and click "Submit" without any modifications.  Alternatively, assert the equality of two different strings using JUnit, as below:
```java
import org.junit.Test;
import static org.junit.Assert.assertEquals;

public class PersonTest {
    @Test
    public void testGreet() {
        assertEquals("Hi Kate, my name is Joe", 
                     "Hi Joe, my name is Kate");
    }
}
```

__Expected behavior__: The JUnit test failure should be displayed properly as something like:
```
testGreet(PersonTest)
  expected: "Hi Kate, my name is Joe" but was: "Hi Joe, my name is Kate"
```

__Buggy behavior__: 
The failure message is displayed as:
![error](https://cloud.githubusercontent.com/assets/5090280/14762632/1ce92e7e-094d-11e6-8494-11fe2b0a89f0.PNG)
Yet the Firefox Web Developer Inspector Tool shows that the expected and actual strings are indeed in the output, but mangled into html markup:
![html](https://cloud.githubusercontent.com/assets/5090280/14762633/225d3026-094d-11e6-83c4-1d4281b668c3.PNG)
